sapi/fpm/fpm/fpm_stdio.c

```diff
@@ -130,10 +132,15 @@ static void fpm_stdio_child_said(struct fpm_event_s *ev, short which, void *arg)
 	} else {
 		event = &child->ev_stderr;
 	}
+	
+	/* calculate how long the "child 123 said into ..." prefix is so we can subtract its length from max_buf_size */
+	fmt_len = snprintf(fmt_buf, max_buf_size, "[pool %s] child %d said into %s: \"%%s\"%%s", child->wp->config->name, (int) child->pid, is_stdout ? "stdout" : "stderr");
+	/* subtract two characters each for the two "%s", then add character counts for "WARNING: " added by zlog and "[11-Feb-2015 15:58:20] " added by zlog; we're ignoring the potential extra length in log_level=DEBUG */
+	fmt_len = fmt_len - 2 - 2 + 9 + 23; 
```

Comment:
But that affects `php_log_err` and `php_error_cb`, @Tyrael. What I'm saying is: instead of truncating with `...` in FPM's `vzlog`, why not do the wrapping to 1024 characters there, and split across multiple lines instead?

The other, really simple solution, is also still an option: reduce `max_buf_size` in `fpm_stdio_child_said`.